### PR TITLE
1.3.1

### PR DESCRIPTION
**Pull request by vrabeshko**
make cache data in filesystem optional if bad permissions

### CODE REVIEW

**Comment by vrabeshko** [comment]
[Cache error in WP plugin / SDK](https://trello.com/c/DmFokr79/1617-cache-error-in-wp-plugin)

File: src/XcooBee/Store/CachedData.php [hunk 1]
```diff
@@ -66,6 +69,10 @@ private static function _getDriver() {
      */
     public function setStore($key, $value)
     {
+        if ($this->_store === null) {
+            return null;
```
Comment (dhurlburtusa) [inline src/XcooBee/Store/CachedData.php]: Can you just return nothing in PHP? This is a setter method. Or is PHP like JavaScript where `undefined` is returned when just using the `return` statement without a value?
```suggestion
            return;
```
If yes, then maybe not return anything in the `clearStore` function below.

File: test/src/XcooBee/Store/CachedDataTest.php [hunk 2]
```diff
@@ -21,13 +21,25 @@ public function testSetStore()
         $this->assertEquals('value', Store::getInstance()->getStore('key'));
     }
 
+    public function testSetStore_StoreIsNull()
+    {
+        $store = new Store();
+        $this->assertEquals(null, $store->getStore('key'));
```
Comment (dhurlburtusa) [inline test/src/XcooBee/Store/CachedDataTest.php]: We are calling `getStore` but function name is `testSetStore_...`. Shouldn't we be making a call to `setStore`? I'm confused.

File: test/src/XcooBee/Store/CachedDataTest.php [hunk 3]
```diff
@@ -38,4 +50,11 @@ public function testClearStore()
         $this->assertNull(Store::getInstance()->getStore('key'));
         $this->assertNull(Store::getInstance()->getStore('key2'));
     }
+
+    public function testClearStore_StoreIsNull()
+    {
+        $store = new Store();
+        $this->assertEquals(null, $store->getStore('key', 'value'));
+        $this->assertEquals(null, $store->getStore('key'));
```
Comment (dhurlburtusa) [inline test/src/XcooBee/Store/CachedDataTest.php]: We have calls to `getStore` but function named `testClearStore_...`. Shouldn't we be making a call to `clearStore`? I'm confused.

File: src/XcooBee/Store/CachedData.php (hunk 1 above)
Comment (Bilal-S) [reply src/XcooBee/Store/CachedData.php]: I would prefer that we are explicit in our returns. Though it should be equivalent.

**Pull request merged**